docarray/array/mixins/io/pushpull.py

```diff
@@ -57,7 +57,9 @@ class PushPullMixin:
 
     _max_bytes = 4 * 1024 * 1024 * 1024
 
-    def push(self, name: str, show_progress: bool = False) -> Dict:
+    def push(
+        self, name: str, *, public: bool = True, show_progress: bool = False
```

Comment:
```suggestion
        self, name: str, show_progress: bool = False, public: bool = True
```
I think better not to change the order and to we don't need to provision extra positional arguments.
If an argument is is necessary it would have existed by now, if there will be one necessary argument in the future, then it means we're breaking the API and no longer supporting the current usage (in that case, we also break the interface). Otherwise, new parameters can be just keyword arguments